### PR TITLE
Syntax

### PR DESCRIPTION
**Pull request by vpetrovykh**
Syntax changes regarding `ROLE` and `ON TARGET DELETE`.

### CODE REVIEW

File: edb/lang/edgeql/ast.py [hunk 1]
```diff
@@ -127,6 +127,14 @@ class Cardinality(s_enum.StrEnum):
     DEFAULT = ''
 
 
+class Cascade(s_enum.StrEnum):
```
Comment (elprans) [inline edb/lang/edgeql/ast.py]: I think `LinkTargetDeleteAction` is a more appropriate name for this enumeration.

File: edb/lang/schema/parser/grammar/declarations.py [hunk 2]
```diff
@@ -904,6 +907,7 @@ def _process_pointerspec(self, p: PointerSpec):
         attributes = []
         policies = []
         properties = []
+        on_delete = []
```
Comment (elprans) [inline edb/lang/schema/parser/grammar/declarations.py]: `on_delete` shouldn't be a list.

File: edb/lang/schema/parser/grammar/declarations.py [hunk 3]
```diff
@@ -914,6 +918,8 @@ def _process_pointerspec(self, p: PointerSpec):
                 properties.append(spec)
             elif isinstance(spec, esast.Policy):
                 policies.append(spec)
+            elif isinstance(spec, esast.OnTargetDelete):
+                on_delete.append(spec)
```
Comment (elprans) [inline edb/lang/schema/parser/grammar/declarations.py]: Can we reuse the enum from `edgeql.ast`?

**Comment by vpetrovykh** [comment]
I've amended the commits to reflect the changes @elprans suggested.

**Comment by elprans** [comment]
LGTM

**Pull request merged**